test_ros2trace/test/test_ros2trace/test_trace.py

```diff
@@ -662,6 +681,76 @@ def test_start_pause_resume_stop(self) -> None:
 
         shutil.rmtree(tmpdir)
 
+    @unittest.skipIf(not are_tracepoints_included(), 'tracepoints are required')
+    def test_snapshot_start_pause_resume_stop(self) -> None:
+        tmpdir = self.create_test_tmpdir('test_snapshot_start_pause_resume_stop')
+        session_name = 'test_snapshot_start_pause_resume_stop'
+
+        # Start tracing and run nodes
+        ret = self.run_trace_subcommand(
+            [
+                'start', session_name,
+                '--ust', tracepoints.rcl_subscription_init, TRACE_TEST_ID_TP_NAME,
+                '--path', tmpdir,
+                '--snapshot-mode',
+            ]
+        )
+        self.assertEqual(0, ret)
+        self.assertSnapshotTracingSessionExist(session_name)
+        self.run_nodes()
+
+        # Pause tracing, record snapshot and check trace
+        ret = self.run_trace_subcommand(['pause', session_name])
+        self.assertEqual(0, ret)
+        ret = self.run_trace_subcommand(['record_snapshot', session_name])
+        self.assertEqual(0, ret)
+        trace_dir = os.path.join(tmpdir, session_name)
+        self.assertTraceExist(trace_dir)
+        self.assertSnapshotTracingSessionExist(session_name)
+        expected_trace_data = [
+            ('topic_name', '/ping'),
+            ('topic_name', '/pong'),
+        ]
+        num_events = self.assertTraceContains(trace_dir, expected_field_value=expected_trace_data)
+
+        # Pausing again should give an error
+        ret = self.run_trace_subcommand(['pause', session_name])
+        self.assertEqual(1, ret)
+
+        # Subbuffers should be cleared after the last record_snapshot
+        ret = self.run_trace_subcommand(['record_snapshot', session_name])
+        self.assertEqual(0, ret)
+        self.assertSnapshotTracingSessionExist(session_name)
+        expected_trace_data = []
+        new_num_events = self.assertTraceContains(
+            trace_dir,
+            expected_field_value=expected_trace_data
+            )
+        self.assertEqual(num_events, new_num_events, 'subbuffers were not cleared')
+
+        # Resume tracing and run nodes again
+        ret = self.run_trace_subcommand(['resume', session_name])
+        self.assertEqual(0, ret)
+        self.assertSnapshotTracingSessionExist(session_name)
+        self.run_nodes()
+
+        # Resuming tracing again should give an error
+        ret = self.run_trace_subcommand(['resume', session_name])
+        self.assertEqual(1, ret)
+        self.assertSnapshotTracingSessionExist(session_name)
+
+        # Stop tracing and check that session does not exist
+        ret = self.run_trace_subcommand(['stop', session_name])
+        self.assertEqual(0, ret)
+        self.assertSnapshotTracingSessionNotExist(session_name)
+
+        # Stopping tracing again should give an error
+        ret = self.run_trace_subcommand(['stop', session_name])
+        self.assertEqual(1, ret)
+        self.assertSnapshotTracingSessionNotExist(session_name)
```

Comment:
Can you call `record_snapshot` after this and check that it properly returns an error/1 since the session (name) doesn't exist?